iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/function_ref.inl

```diff
@@ -15,15 +15,34 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
+#ifndef IOX_HOOFS_CXX_FUNCTION_REF_INL
+#define IOX_HOOFS_CXX_FUNCTION_REF_INL
+
+#include "iceoryx_hoofs/cxx/function_ref.hpp"
+#include "iceoryx_hoofs/cxx/requires.hpp"
+
+#include <memory>
+
 namespace iox
 {
 namespace cxx
 {
 template <class ReturnType, class... ArgTypes>
 template <typename CallableType, typename>
+// AXIVION Next Line AutosarC++19_03-A12.1.2 : Members are initialized in the same manner, NSDMI with nullptr is
+// explicit
+// AXIVION Next Line AutosarC++19_03-A8.4.6 : Only ArgTypes needs to be forwarded
 inline function_ref<ReturnType(ArgTypes...)>::function_ref(CallableType&& callable) noexcept
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4, AutosarC++19_03-A5.2.3, CertC++-EXP55 : Type-safety ensured by
+    // casting back on call
+    // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast, cppcoreguidelines-pro-type-const-cast)
     : m_pointerToCallable(const_cast<void*>(reinterpret_cast<const void*>(std::addressof(callable))))
+    // AXIVION Next Line AutosarC++19_03-A15.4.4 : Lambda not 'noexcept' as callable might throw
```

Comment:
With this argument the call operator must be `noexcept`.

Reply:
@elfenpiff I don't get this, the `operator()()` is already `noexcept`.

@MatthiasKillat Correct, we need to monitor that.